src/uucore/src/lib/features/process.rs

```diff
@@ -99,3 +131,17 @@ impl ChildExt for Child {
         Ok(None)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    #[cfg(not(target_os = "redox"))]
+    fn test_getsid() {
+        assert!(getsid(getpid()).is_ok());
```

Comment:
How about comparing it with `getsid(0)`? This would also make sure that the result is semi-reasonable, consistent, and would also check that code path.

```suggestion
        assert_eq!(getsid(getpid()).expect("getsid(getpid)"), getsid(0).expect("getsid(0)"));
```

Reply:
Can you explain this in more detail please? It seems like the first time I've seen this used, and it seems like an interesting trick but I'm slightly confused XD

Reply:
Sure, thanks for asking!

Currently, your test calls `getsid` with the own PID. That should return the SID of the current process. So far so good.

However, it's hard to tell whether it's working correctly. What if this returns 42, is that correct, is it wrong? How would you even detect that `getsid` is broken, assuming it is broken? (After all, that is the entire purpose of testing!)

So I'm suggesting to call `getsid` in a slightly different way, and make sure that `getsid()` at least returns consistent data: `getsid(0) == getsid(getpid())`
This is because `0` is a special value for `getsid`: https://pubs.opengroup.org/onlinepubs/9699919799/functions/getsid.html

However, this equality-check doesn't compile, because `Result<_, Error>` does not implement `trait PartialEq`. That's why I'm calling `.expect("…")`, which is basically the same thing as `.unwrap()`: https://doc.rust-lang.org/std/io/type.Result.html#method.expect

I just notice something new: The SID must never be zero, so we might add something like `assert!(sid > 0);`
Not sure whether it really adds anything of value, but it's cheap enough to just throw it in.

What do you think?

Reply:
So amazing!